### PR TITLE
Fix for issue #48

### PR DESCRIPTION
**Pull request by herbig**
Resolves issue #48

### CODE REVIEW

File: app/src/main/java/org/standardnotes/notes/frag/NoteFragment.kt [hunk 1]
```diff
@@ -46,6 +47,11 @@ class NoteFragment : Fragment(), SyncManager.SyncListener {
         }
     }
 
+    override fun onSaveInstanceState(outState: Bundle) {
```
Comment (deftelf) [inline app/src/main/java/org/standardnotes/notes/frag/NoteFragment.kt]: This is a good call - though I don't see where it's being retrieved...

**Comment by deftelf** [comment]
I'm a bit unclear as what this achieves, given the conversation in #48 ... What changes are we expecting to see propagate from other open clients?

**Comment by herbig** [comment]
My understanding is that if a change is made on, say the web client, for a note that is currently open on the app, the app should show that update.

**Comment by mobitar** [comment]
Yes, for notes title and text, text input should update if a change comes in. This can happen if the user closes the app while a note is open, comes back a day later, and expects to see changes he made somewhere else come in. One user already emailed me about being bit by this.

**Comment by mobitar** [comment]
But maybe it's just how the Android app is set up, but this seems very manual:

```
for (retrievedItem in syncItems.retrievedItems) {
 +            if (retrievedItem.uuid == note?.uuid) {
 +                if (retrievedItem.updatedAt > note?.updatedAt) {
 +                    note = SApplication.instance!!.noteStore.getNote(retrievedItem.uuid)
 +                    tags = SApplication.instance!!.noteStore.getTagsForNote(retrievedItem.uuid)
 +                    populateUI()
 +                }
 +                break
 +            }
 +        }
```

Are object instances not just passed around from view to view? So if the sync manager receives changes and updates an object reference, shouldn't that update all references of the object no matter which view it's in?

In iOS, I just add an observer on the "title" and "text" properties of the object to detect when a change is made.

**Comment by deftelf** [comment]
Not generally no, as separate screens can be reclaimed for memory so they don't share data by default.
I'm not immediately convinced it would be a good idea, gonna have a mull over it. I've found objects that get edited from multiple locations can cause subtle bugs

**Comment by mobitar** [comment]
Wait so it's not customary to pass one object from one view to another in Android? That's extremely strange.

In any case, shouldn't there be some sort of singleton that handles, and is the root authority for, all models? This way, views look to the singleton for data changes. And views can subscribe to model changes with this singleton. But there has to be a better way then looping through synced objects and finding the one you want.

But you're right, this probably isn't a priority for 1.0, but can be a quick follow.

**Comment by herbig** [comment]
Yeah, I agree with @deftelf, I don't think it's common to keep a set of objects statically in memory. Also, passing whole objects between screens, via bundles is usually not a good method, and will for sure introduce syncing bugs.

In this case we do have the database Notestore, which is the authority.  It requires a database query when opening a note, but we can assume that whatever is in the database is what is current.

For determining whether or not a currently opened note has been updated from a different source, we need to know which array it was in when it was received, which is not a property of the note itself, hence looping through, which is definitely awkward looking.

**Comment by mobitar** [comment]
Interesting. So when the app launches, are all notes not loaded from the database and into memory?

**Comment by deftelf** [comment]
They are - they probably shouldn't be actually, given how big they might all be. 

Incidentally, I don't think that the immutable(ish) objects model is particularly unusual.

I am holding off merging this until I've worked out why we are getting crashes on big dbs, btw.

**Comment by deftelf** [comment]
The NoteStore can manage memory cached versions of the objects if necessary, and do the shareing and change events if we consider that a worthy architecture move.

**Comment by mobitar** [comment]
iOS currently loads everything into memory. As this becomes a problem, we'll work on optimizations, but it's important to just have a starting point. So not really a priority to re-think architecture.

**Comment by deftelf** [comment]
This is actually resolved by the upcoming tablet refactor.

**Pull request closed without merging**